ssl/statem/statem_srvr.c

```diff
@@ -3593,6 +3634,21 @@ MSG_PROCESS_RETURN tls_process_client_certificate(SSL *s, PACKET *pkt)
     return ret;
 }
 
+#if !defined(OPENSSL_NO_COMP)
+MSG_PROCESS_RETURN tls_process_client_compressed_certificate(SSL *s, PACKET *pkt)
+{
+    MSG_PROCESS_RETURN ret = MSG_PROCESS_ERROR;
+    PACKET tmppkt;
+    BUF_MEM *buf = BUF_MEM_new();
```

Comment:
This allocation can failed.

Reply:
Yes, and it's checked in `tls13_process_compressed_certificate`; which is called by both the client and server code. So, I put the check in there, rather than having two checks before the call of the function.